src/penn_chime/models.py

```diff
@@ -72,187 +77,181 @@ def __init__(self, p: Parameters):
         raw_df = sim_sir_df(
             susceptible,
             infected,
-            recovered,
+            p.recovered,
             beta,
             gamma,
             p.n_days,
         )
-        dispositions_df = build_dispositions_df(raw_df, rates, p.market_share)
-        admits_df = build_admits_df(dispositions_df, p.n_days_since_first_hospitalized)
-        census_df = build_census_df(admits_df, lengths_of_stay, p.n_days_since_first_hospitalized)
+        dispositions_df = build_dispositions_df(raw_df, rates, p.market_share, p.current_date)
+        admits_df = build_admits_df(dispositions_df)
+        census_df = build_census_df(admits_df, lengths_of_stay)
 
         self.susceptible = susceptible
         self.infected = infected
-        self.recovered = recovered
+        self.recovered = p.recovered
 
         self.detection_probability = detection_probability
-        self.recovered = recovered
-        self.intrinsic_growth_rate = intrinsic_growth_rate
-        self.gamma = gamma
+        self.doubling_time_t = doubling_time_t
         self.beta = beta
+        self.gamma = gamma
+        self.intrinsic_growth_rate = intrinsic_growth_rate
         self.r_t = r_t
         self.r_naught = r_naught
-        self.doubling_time_t = doubling_time_t
-        self.raw_df = raw_df
-        self.dispositions_df = dispositions_df
-        self.admits_df = admits_df
-        self.census_df = census_df
 
-        if p.n_days_since_first_hospitalized is None and p.doubling_time is not None:
-            p.n_days_since_first_hospitalized = int(self.argmin_ds(p))
-            p.date_first_hospitalized = date.today() - timedelta(days=p.n_days_since_first_hospitalized)
-            admits_df = build_admits_df(dispositions_df, p.n_days_since_first_hospitalized)
-            census_df = build_census_df(admits_df, lengths_of_stay, p.n_days_since_first_hospitalized)
-            self.admits_df = admits_df
-            self.census_df = census_df
-        if p.n_days_since_first_hospitalized is not None and p.doubling_time is None:
-            # optimize doubling_time
-            argmin_dt = None
+        if p.date_first_hospitalized is None and p.doubling_time is not None:
+            print('%s: use doubling_time.' % (datetime.now(),))
```

Comment:
same question- is this debug output or do you have a user story like "i cloned and am running the app locally to generate reports, i like to read these details" etc?

Reply:
I'll add a logger